src/mame/bfm/bfm_blackbox.h

```diff
@@ -0,0 +1,299 @@
+// license:BSD-3-Clause
+// copyright-holders:SomeRandomGuyIdk
+/**********************************************************************
+
+	Bellfruit Black Box
+
+**********************************************************************/
+
+#ifndef MAME_INCLUDES_BFM_BLACKBOX_H
+#define MAME_INCLUDES_BFM_BLACKBOX_H
+
+#pragma once
+
+#include "fruitsamples.h"
+
+#include "cpu/m6800/m6800.h"
+#include "cpu/tms1000/tms1000.h"
+#include "machine/6821pia.h"
+#include "machine/em_reel.h"
+#include "machine/timer.h"
+#include "sound/beep.h"
+#include "sound/spkrdev.h"
+
+#define OUT_READ(name) uint8_t name##_r(offs_t offset) { name##_w(offset, 0); return 0; }
+#define OUT_READ_T(name) template <unsigned T> uint8_t name##_r(offs_t offset) { name##_w<T>(offset, 0); return 0; }
+#define OUTPUT(name) rw(FUNC(name##_r), FUNC(name##_w))
+#define OUTPUT_T(name, T) rw(FUNC(name##_r<T>), FUNC(name##_w<T>))
```

Comment:
Please don’t do this.  It just obfuscates things and makes debugging more difficult.  We’re trying to move aware from macros except in cases where they’re really needed.  We already demacrofied all the address maps.  This is regressive.

Reply:
Fixed.